### ISSUE TITLE
term_start crash in windows

### ISSUE THREAD

**Issue by wsdjeg** [vim/vim#2502]
if $SHELL is empty, then:

```vim
call term_start($SHELL, {'curwin' : 1, 'term_finish' : 'close'})
```

cause vim crash in windows

**Comment by wsdjeg** [comment]
![qq 20171227085526](https://user-images.githubusercontent.com/13142418/34367287-b36a4a12-eae3-11e7-8f56-96133e8fdd8e.png)

**Comment by chrisbra** [comment]
you shouldn't use $SHELL, since this is not defined in windows. Better would be to use `&sh` (note you might need to escape spaces in there).

**Comment by mattn** [comment]
Yes, we know. This is a bug that vim crash with empty argumentfor term_start.

**Comment by k-takata** [comment]
Fixed in [8.0.1429](https://github.com/vim/vim/commit/ede35bbbd05306097bf3f4d603f2248ed1f4c5f1).

**Comment by wsdjeg** [comment]
Thanks k-takata
